### PR TITLE
Honor contexts passed to registry client methods

### PR DESCRIPTION
**Pull request by 2opremio**
We noticed that some requests passed to the repository client were possibly hanging https://github.com/weaveworks/flux/issues/1940 . 

One of the causes is that the client doesn't honor the contexts passed to it https://github.com/weaveworks/flux/issues/1940#issuecomment-484738754 . This PR fixes that.

### CODE REVIEW

**Comment by carnott-snap** [comment]
This change came up as a blocker for some improvements I am working on for fluxcd/flux; is there any outstanding work required?

cc: @aaronlehmann @dmcgowan @RichardScothern

File: registry/client/repository.go [hunk 1]
```diff
@@ -307,7 +307,7 @@ func (t *tags) Get(ctx context.Context, tag string) (distribution.Descriptor, er
 	}
 
 	newRequest := func(method string) (*http.Response, error) {
-		req, err := http.NewRequest(method, u, nil)
+		req, err := newRequestWithContext(ctx, method, u, nil)
```
Comment (dmcgowan) [inline registry/client/repository.go]: Now the method can be used from the `http` package

Comment (2opremio) [reply registry/client/repository.go]: I don't think I follow

Comment (dmcgowan) [reply registry/client/repository.go]: https://golang.org/pkg/net/http/#NewRequestWithContext

Also for the other cases, the helper functions shouldn't be necessary after this function is used

Comment (2opremio) [reply registry/client/repository.go]: Gotcha. The code is now using `http.NewRequestWithContext()`.

I also removed the auxiliary functions but I think it looked cleaner with them.

**Comment by michaelbeaumont** [comment]
@dmcgowan Hi, is there a chance of getting a review on this PR?

**Review by milosgajdos — APPROVED** [review]

**Review by joaodrp — APPROVED** [review]

**Pull request merged**